examples/training/object_detection/pascal_voc/retina_net.py

```diff
@@ -202,41 +253,85 @@ def apply(inputs):
         bounding_boxes = keras_cv.bounding_box.convert_format(
             bounding_boxes, images=image, source="yxyx", target=bounding_box_format
         )
-        return {"images": image, "bounding_boxes": bounding_boxes}
+        return image, bounding_boxes
 
     return apply
 
 
+def proc_eval_fn(bounding_box_format, target_size):
+    def apply(inputs):
+        raw_image = inputs["image"]
+        raw_image = tf.cast(raw_image, tf.float32)
+
+        img_size = tf.shape(raw_image)
+        height = img_size[0]
+        width = img_size[1]
+
+        target_height = tf.cond(
+            height > width,
+            lambda: 640.0,
+            lambda: tf.cast(height / width * 640.0, tf.float32),
+        )
+        target_width = tf.cond(
+            width > height,
+            lambda: 640.0,
+            lambda: tf.cast(width / height * 640.0, tf.float32),
+        )
+        image = tf.image.resize(
+            raw_image, (target_height, target_width), antialias=False
+        )
+
+        gt_boxes = keras_cv.bounding_box.convert_format(
+            inputs["objects"]["bbox"],
+            images=image,
+            source="rel_yxyx",
+            target="xyxy",
+        )
+        image = tf.image.pad_to_bounding_box(
+            image, 0, 0, target_size[0], target_size[1]
+        )
+        gt_boxes = keras_cv.bounding_box.convert_format(
+            gt_boxes,
+            images=image,
+            source="xyxy",
+            target=bounding_box_format,
+        )
+        gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
+        gt_classes = tf.expand_dims(gt_classes, axis=-1)
+
+        bounding_boxes = tf.concat([gt_boxes, gt_classes], axis=-1)
+
+        return image, bounding_boxes
+
+    return apply
+
+
+def pad_fn(image, boxes):
+    return image, boxes.to_tensor(default_value=-1.0)
+
+
 train_ds = train_ds.map(
     proc_train_fn("xywh", image_size), num_parallel_calls=tf.data.AUTOTUNE
 )
 train_ds = train_ds.apply(
     tf.data.experimental.dense_to_ragged_batch(BATCH_SIZE, drop_remainder=True)
```

Comment:
Shouldn't we use the global batch size here for the original dataset?

Reply:
Good catch. Done